mapadroid/worker/MITMBase.py

```diff
@@ -55,123 +56,83 @@ def __init__(self, args, dev_id, origin, last_known_state, communicator: Abstrac
+        self.logger.info('Waiting for data after {}',
+                         datetime.fromtimestamp(timestamp))
+        position_type = self._mapping_manager.routemanager_get_position_type(self._routemanager_name,
+                                                                             self._origin)
+        type_of_data_returned = LatestReceivedType.UNDEFINED
+        data = None
+        while type_of_data_returned == LatestReceivedType.UNDEFINED and \
+                int(timestamp + timeout) >= int(time.time()) \
                 and not self._stop_worker_event.is_set():
             latest = self._mitm_mapper.request_latest(self._origin)
+
+            if latest is None:
+                self.logger.debug("Nothing received from worker since MAD started")
+                time.sleep(0.5)
+                continue
+            # Not checking the timestamp against the proto awaited in here since custom handling may be adequate.
+            # E.g. Questscan may yield errors like clicking mons instead of stops - which we need to detect as well
             latest_location: Optional[Location] = latest.get("location", None)
             check_data = True
             if (proto_to_wait_for == 106 and latest_location is not None and
-                    latest_location.lat != 0.0 and latest_location.lng != 0.0):
-                self.logger.debug("Checking worker location {} against real data location {}", self.current_location,
+                    latest_location.lat != 0.0 and latest_location.lng != 0.0 and
+                    -90.0 <= latest_location.lat <= 90.0 and
+                    -180.0 <= latest_location.lng <= 180.0):
+                self.logger.debug("Checking GMO location reported by {} at {} against real data location {}",
+                                  self._origin,
+                                  self.current_location,
                                   latest_location)
                 distance_to_data = get_distance_of_two_points_in_meters(float(latest_location.lat),
                                                                         float(latest_location.lng),
                                                                         float(self.current_location.lat),
                                                                         float(self.current_location.lng))
                 max_distance_of_mode = self._mapping_manager.routemanager_get_max_radius(self._routemanager_name)
-                max_distance_for_worker: int = self._applicationArgs.maximum_valid_distance \
-                    if max_distance_of_mode < self._applicationArgs.maximum_valid_distance else max_distance_of_mode
+                max_distance_for_worker = self._applicationArgs.maximum_valid_distance
+                if max_distance_for_worker > max_distance_of_mode > 5:
+                    # some modes may be too strict (e.g. quests with 0.0001m calculations for routes)
```

Comment:
We should have a variable that is descriptive of why there is a minimum of five.